### PR TITLE
failover doc

### PR DESCRIPTION
**Pull request by yuval-k**

### CODE REVIEW

File: design-docs/global-services.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  hosts:
+  - reviews-global.default.global
+  location: MESH_INTERNAL
+  ports:
+  - name: http1
+    number: 9080
+    protocol: http
+  resolution: STATIC
+  addresses:
+  - 240.0.0.20
```
Comment (harveyxia) [inline design-docs/global-services.md:67]: How is this static IP address determined?

File: design-docs/global-services.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        connect_timeout: 1s
+        lb_policy: CLUSTER_PROVIDED
+        cluster_type:
+          name: envoy.clusters.aggregate
+          typed_config:
+            "@type": type.googleapis.com/udpa.type.v1.TypedStruct
+            type_url: type.googleapis.com/envoy.config.cluster.aggregate.v2alpha.ClusterConfig
+            value:
+              clusters:
+              - outbound|9080||reviews.default.svc.cluster.local
```
Comment (harveyxia) [inline design-docs/global-services.md:103]: Just to confirm, is this list ordered by decreasing priority?

File: design-docs/global-services.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      interval: 5s
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+```
+
+# Remote cluster
+On the remote cluster we will configure these objects, to redirect incoming traffic to the correct local service (pending final testing):
+```
+apiVersion: networking.istio.io/v1beta1
```
Comment (harveyxia) [inline design-docs/global-services.md:129]: Could we also use an `EnvoyFilter` to perform this redirection? That's how [service federation](https://github.com/solo-io/service-mesh-hub/blob/master/services/mesh-networking/pkg/federation/resolver/meshes/istio/istio_federation_client.go#L441) is currently working.

What are the pros/cons, if any, to using a VirtualService vs. EnvoyFilter for this?

Comment (yuval-k) [reply design-docs/global-services.md:129]: Better to use vs if you can. EnvoyFilter is a break glass configuration that should be avoided if possible

File: design-docs/global-services.md (hunk 2 above)
Comment (yuval-k) [reply design-docs/global-services.md:103]: Correct

File: design-docs/global-services.md (hunk 1 above)
Comment (yuval-k) [reply design-docs/global-services.md:67]: It is decided. 
It just needs to be unique.  It doesn't need to be a real ip address

Comment (yuval-k) [reply design-docs/global-services.md:67]: In practice will probably scan existing sevice entries to generate a unique one

Comment (EItanya) [reply design-docs/global-services.md:67]: We already have a component for this that @GrahamGoudeau designed
https://github.com/solo-io/service-mesh-hub/blob/master/services/mesh-networking/pkg/federation/dns/ip_assigner.go

File: design-docs/global-services.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+spec:
+  selector:
+    istio: ingressgateway # use istio default ingress gateway
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - reviews-global.default.global
```
Comment (christian-posta) [inline design-docs/global-services.md:164]: Do we need this `Gateway` resource? Could we just match on all *.global services?

File: design-docs/global-services.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+we need it so `reviews-global.default.global` is resolvable.
+
+```
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: reviews-global
+spec:
+  hosts:
+  - reviews-global.default.global
```
Comment (christian-posta) [inline design-docs/global-services.md:59]: Do we need to specify the namespace in the name? For global services, the namespace may not be consistent across clusters, so doesn't really mean much. Unless we're trying to create a global namespace for naming conventions that's not tied to Kubernetes namespaces?

**Pull request merged**